### PR TITLE
Use "-pthread" for GoogleTest builds in Emscripten

### PR DESCRIPTION
**Pull request by emaxx-google**
Add the "-pthread" flag when compiling and linking the GoogleTest static
libraries in Emscripten builds.

Meanwhile Pthreads aren't actually used by GoogleTest itself (due to the
"gtest_disable_pthreads" parameter that we specify), it's still necessary
because we use "-pthread" when compiling other libraries and source
files, and Emscripten seems to require this flag to be used consistently
everywhere.

This is a preparatory fix for introducing Emscripten builds of unit
tests, as tracked by #177.

### CODE REVIEW

File: third_party/googletest/webport/build/Makefile [hunk 1]
```diff
@@ -78,14 +78,24 @@ endif
 #   the compilation according to prepared scripts.
 #
 # Explanation of parameters to cmake:
-# B: build directory.
+# E env: Wraps the further second cmake call into the "env" tool that allows to
```
Comment (Fabian-Sommer) [inline third_party/googletest/webport/build/Makefile:81]: nit: Isn't it the first cmake call, rather than the second? Also, delete "further".

**Review by Fabian-Sommer — APPROVED** [review]
LGTM % nit

File: third_party/googletest/webport/build/Makefile (hunk 1 above)
Comment (emaxx-google) [reply third_party/googletest/webport/build/Makefile:81]: The first `cmake E env ...` call is used to set up the environment values for the second `cmake ../../src` one. If the comment doesn't do a good job in explaining that, I'm happy to rephrase it.

Comment (Fabian-Sommer) [reply third_party/googletest/webport/build/Makefile:81]: I see. I think I would go with `Wraps the succeeding cmake call`

**Review by MahmoudAGawad — APPROVED** [review]
LGTM.

**Comment by emaxx-google** [comment]
Thanks!

**Pull request merged**